app/models/authenticator.rb

```diff
@@ -247,11 +247,12 @@ def run_task(taskid, status)
       end
     end
 
+    # TODO: Fix this icky select matching with tenancy
     def match_groups(external_group_names)
       return [] if external_group_names.empty?
       external_group_names = external_group_names.collect(&:downcase)
 
-      internal_groups = MiqServer.my_server.permitted_groups
+      internal_groups = MiqGroup.order(:sequence).all.to_a
```

Comment:
This is gross but is what is currently being done by `permitted_groups` anyway. Will likely be back in here shortly with tenancy work to make this betterererer.